common/lib/xmodule/xmodule/course_module.py

```diff
@@ -269,6 +269,44 @@ def get_available_providers():
     return available_providers
 
 
+class DiscussionTopicMapping(Dict):
+    """
+    DiscussionTopicMapping field, which includes validation of the topic id
+    """
+
+    _default = {}
+
+    def from_json(self, value):
+        """
+        Return python dict type. Performs validation on id key for each discussion topic
+        """
+        if value is None:
+            return None
+        """
+        if statement checks to make sure value is a dict,
+        each topic in dict has a key id and the id value only has valid characters
+        """
+        if isinstance(value, dict):
+            for key in value:
+                if "id" in value[key].keys():
+                    if not self._validate_topic_id_value(value[key]["id"]):
+                        raise ValueError("The only special characters that are supported in the id key are underscore, hyphen, and period.")
```

Comment:
Can we make this a bit more descriptive to note that it's characters supported for discussion IDs? Also will this be seen by users or is this strictly for our logs?

Reply:
This will be seen by the users